proto/gossip.proto

```diff
@@ -44,6 +46,61 @@ message GossipRequest {
 message GossipResponse {
   // Requested delta of server's infostore.
   optional bytes delta = 1;
+  // TODO(thschroeter): replace bytes with GossipInfoStore
+  //optional GossipInfoStore delta = 1;
   // Non-nil means client should retry with this address.
   optional Addr alternate = 2;
 }
+
+message GossipInfoStore {
+  // Map from key to info
+  map<string, GossipInfo> infos = 1;
```

Comment:
I didn't know this before but the version 3.x proto compiler enables some of the new features even in `proto2` mode (https://groups.google.com/d/msg/protobuf/p4WxcplrlA4/vJXyICvjEgAJ). It's unclear what exactly this covers, but as long as the versions of protoc and gogoproto that we're pinning in the GLOCKFILE support maps we might as well use them (at least for internal-only protos. We should probably be more conservative in client-facing protos)